packages/metro-minify-terser/src/minifier.js

```diff
@@ -50,11 +50,7 @@ function minify({
   /* $FlowFixMe(>=0.111.0 site=react_native_fb) This comment suppresses an
    * error found when Flow v0.111 was deployed. To see the error, delete this
    * comment and run Flow. */
-  const result = terser.minify(code, options);
-
-  if (result.error) {
```

Comment:
Terser will now throw an error so there is no need for additional handling anymore.